main.tf

```
@@ -1,3 +1,17 @@
-resource "random_pet" "template" {
-  length = var.length
+locals {
+  users = { for u in var.users : u.username => u }
+  roles = {
+    "standard"  = data.datadog_role.standard.id
+    "admin"     = data.datadog_role.admin.id
+    "read_only" = data.datadog_role.read_only.id
+  }
+}
+
+resource "datadog_user" "users" {
+  for_each             = local.users
+  disabled             = each.value.disabled
+  email                = each.value.email
+  name                 = each.value.name
+  roles                = [local.roles[each.value.access_roles["datadog"].role]]
```

In line with my comment below: 

```suggestion
  roles                = [local.roles[each.value.role]]
```